broker-core/src/main/java/io/zeebe/broker/incident/processor/IncidentStreamProcessor.java

```diff
@@ -54,12 +55,16 @@
   private final Long2LongZbMap resolvingEvents = new Long2LongZbMap();
 
   public TypedStreamProcessor createStreamProcessor(TypedStreamEnvironment env) {
+    final KeyGenerator keyGenerator = KeyGenerator.createIncidentKeyGenerator();
+
     TypedEventStreamProcessorBuilder builder =
         env.newStreamProcessor()
+            .keyGenerator(KeyGenerator.createIncidentKeyGenerator())
             .withStateResource(activityInstanceMap)
             .withStateResource(failedJobMap)
             .withStateResource(incidentMap.getMap())
-            .withStateResource(resolvingEvents);
+            .withStateResource(resolvingEvents)
+            .withStateResource(keyGenerator);
```

Comment:
You don't need to register the `KeyGenerator` as a state resource. This is already done by setting it.